sklearn/preprocessing/_data.py

```diff
@@ -2481,10 +2481,11 @@ class QuantileTransformer(OneToOneFeatureMixin, TransformerMixin, BaseEstimator)
         matrix are discarded to compute the quantile statistics. If False,
         these entries are treated as zeros.
 
-    subsample : int, default=10_000
+    subsample : int or None, default=10_000
         Maximum number of samples used to estimate the quantiles for
         computational efficiency. Note that the subsampling procedure may
         differ for value-identical sparse and dense matrices.
+        Setting `subsample=None` disables subsampling.
```

Comment:
I'd reverse the sentence to put "disable subsampling" at the front, making it easier to find for people who are scanning the docs (same for the doc string of the function later one)
```suggestion
        Disable subsampling with `subsample=None`.
```
(or something like this)